### PR TITLE
A shed load of changes

### PR DESCRIPTION
**Pull request by steveww**
Removed all the old Instana stuff
Reimplemented rating service in Python. No PHP expertise.
Tidy up Service Monitors
Health and Ready probes
Numerous other tweaks.

### CODE REVIEW

File: cart/server.js [hunk 1]
```diff
@@ -117,44 +130,50 @@ app.delete('/cart/:id', (req, res) => {
             throw new Error('Invalid cart id: ' + req.params.id);
         }
     } catch(e) {
-        logger.error(e);
-        res.status(500).send(e.message);
+        req.log.error(e);
```
Comment (nandadev) [inline cart/server.js:133]: Let's retain the `logger.error(e)` to navigate to the cart error logs (in Loki) from Workbench

File: cart/server.js [hunk 2]
```diff
@@ -381,49 +394,44 @@ function calcTax(total) {
...
-                resolve(data);
-            }
-        });
-    });
+    logger.info('saving cart %o', cart);
+    return redisClient.SETEX(id, 3600, JSON.stringify(cart));
 }
 
 // connect to Redis
+logger.info('Connectiong to redis host %s', redisHost);
```
Comment (nandadev) [inline cart/server.js:417]: typo in Connection

File: K8s/helm/templates/rabbitmq-service-monitor.yaml [hunk 3]
```diff
@@ -1,15 +0,0 @@
-apiVersion: monitoring.coreos.com/v1
-kind: ServiceMonitor
-metadata:
-  labels:
-    release: kube-prometheus-stack
-  name: rabbitmq
-spec:
-  endpoints:
-    - path: /metrics
-      targetPort: 15692
```
Comment (nandadev) [inline K8s/helm/templates/rabbitmq-service-monitor.yaml:10]: The name of this port is `http-management`. Shouldn't this port name be added to `service-monitor.yaml` so that prom is able to scrape mongodb rabbitmq metrics?

File: load-gen/load-gen-dep.yaml [hunk 4]
```diff
@@ -0,0 +1,30 @@
+apiVersion: apps/v1
```
Comment (nandadev) [inline load-gen/load-gen-dep.yaml:1]: There's already a `load-deployment.yaml` file inside `K8S` folder

File: .env [hunk 5]
```diff
@@ -1,5 +1,5 @@
 # environment file for docker-compose
-REPO=asserts
+REPO=steveww
```
Comment (nandadev) [inline .env:2]: Revert this change?

File: cart/server.js (hunk 1 above)
Comment (steveww) [reply cart/server.js:133]: That looks like a typo on my part. Should have been `req.log` and not `logger`.

File: load-gen/load-gen-dep.yaml (hunk 4 above)
Comment (steveww) [reply load-gen/load-gen-dep.yaml:1]: This is integrated with the scripting to allow for run time configuration. The one under K8s is all hard coded.

File: .env (hunk 5 above)
Comment (steveww) [reply .env:2]: Oops! That's for my local testing. Forgot to revert before pushing.

File: K8s/helm/templates/rabbitmq-service-monitor.yaml (hunk 3 above)
Comment (steveww) [reply K8s/helm/templates/rabbitmq-service-monitor.yaml:10]: Still some work to do on Service Monitor. There's numerous endpoints and ports.

File: cart/server.js (hunk 1 above)
Comment (steveww) [reply cart/server.js:133]: This is inside the `delete` handler.

`req.log` is the same as `logger` both write the log to the same place.
`req.log` is an express wrapped version of logger. See from line 24.

File: K8s/helm/templates/rabbitmq-service-monitor.yaml (hunk 3 above)
Comment (steveww) [reply K8s/helm/templates/rabbitmq-service-monitor.yaml:10]: Adding `prometheus-metrics` to service monitor port definitions.

**Comment by steveww** [comment]
Followed up on your comments. New commits done.

**Review by nandadev — APPROVED** [review]
Changes LGTM
